trpcultivate_phenotypes/tests/src/Kernel/Validators/ValidatorBaseTest.php

```diff
@@ -587,4 +588,80 @@ public function testSplitRowIntoColumnsMultiDelimiter() {
       'Expected exception message does not match message when splitRowIntoColumns() could not split line because there are too many supported delimiters.'
     );
   }
+
+  /**
+   * Tests the ValidatorBase::setLogger() setter
+   *       and ValidatorBase::getLogger() getter
+   *
+   * @return void
+   */
+  public function testTripalLoggerGetterSetter() {
+    $configuration = [];
+    $validator_id = 'fake_basically_base';
+    $plugin_definition = [
+      'id' => $validator_id,
+      'validator_name' => 'Basically Base Validator',
+      'input_types' => ['header-row', 'data-row'],
+    ];
+    $instance = new BasicallyBase($configuration, $validator_id, $plugin_definition);
+    $this->assertIsObject(
+      $instance,
+      "Unable to create fake_basically_base validator instance to test the base class."
+    );
+
+    // Try to get the logger before it has been set
+    // Exception message should trigger
+    $expected_message = 'Cannot retrieve the Tripal Logger property as one has not been set for this validator using the setLogger() method.';
+    $exception_caught = FALSE;
+    $exception_message = 'NONE';
+    try {
+      $instance->getLogger();
+    } catch (\Exception $e) {
+      $exception_caught = TRUE;
+      $exception_message = $e->getMessage();
+    }
+
+    $this->assertTrue($exception_caught, 'Calling getLogger() before the setLogger() method should have thrown an exception but did not.');
+    $this->assertStringContainsString(
+      $expected_message,
+      $exception_message,
+      "The exception thrown does not have the message we expected when trying to get the Tripal Logger property but it hasn't been set yet."
+    );
+
+    // Create a TripalLogger object and set it using setLogger()
+    $my_logger = \Drupal::service('tripal.logger');
+
+    $exception_caught = FALSE;
+    $exception_message = 'NONE';
```

Comment:
None of the cases from this line to the closing bracket checks this value, either drop this or implement an assert equals/containsString.

Reply:
Thanks for catching this! I opted to drop the variable (+ the one on line 646) since these cases don't expect an exception to get thrown (and the actual exception message does get printed if triggered).